### ISSUE TITLE
[MDEPLOY-39] Deploy doesn't recognize repositoryId when using deploy:deploy-file and scp

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-deploy-plugin#277]
**[euvitudo](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=euvitudo)** opened **[MDEPLOY-39](https://issues.apache.org/jira/browse/MDEPLOY-39?redirect=false)** and commented

Maven 2.0.4
maven-deploy-plugin 2.2.1

My settings.xml:

\<settings>
\<servers>
\<server>
\<id>nsa-deploy-repository\</id>
\<username>pwarner\</username>
\<privateKey>/home/pwarner/.ssh/identity\</privateKey>
\<filePermissions>664\</filePermissions>
\<directoryPermissions>775\</directoryPermissions>
\</server>
\</servers>
\</settings>

My command-line:

mvn deploy:deploy-file \
-Dfile=/home/pwarner/jboss/jargon_v1.4.22/lib/jargon.jar \
-Dpackaging="jar" \
-DrepositoryID="nsa-deploy-repository" \
-Durl="scp://chive.tuc.noao.edu/u1/www/htdocs/maven/" \
-DgroupId="sdsc" \
-DartifactId="jargon" \
-Dversion="1.4.22"

Whenever I use the above configuration, it asks for a password, despite the fact that (1) the repositoryId I am using on the command line is configured in the settings.xml, and (2) I specify the location of the ssh key.

I know my key is not bad (it's used when I use the command-line ssh).

But, when I change the repository id (in settings.xml) to the default (remote-repository - as indicated on the last bullet at the bottom of http://maven.apache.org/plugins/maven-deploy-plugin/usage.html ), then I don't have this problem, and the key is used.

I've also tried providing a user/pass in the settings that are different from mine (but that I know are correct) and I get the same results as above.  I.e., when I change the id to remote-repository, it works.

One other thing: after changing the id in settings.xml to remote-repository, and use a different repositoryId on the command line, e.g., nsa-deploy-repository, it works when it really shouldn't.

So, in summary, the repositoryId on the command line is not being honored for deploy:deploy-file.

---

**Affects:** 2.2.1

1 votes, 0 watchers

**Comment by jira-importer** [comment]
**[Brett Porter](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=brettporter)** commented

just to check - are you using repositoryId or repositoryID (as in your example)?

**Comment by jira-importer** [comment]
**[euvitudo](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=euvitudo)** commented

Yeesh... I can't believe I didn't catch that.  Indeed, the problem was that I was using repositoryID.
Sorry for the trouble.
